agent/functional_tests/tests/functionaltests_windows_test.go

```diff
@@ -435,6 +440,248 @@ func TestGMSAFile(t *testing.T) {
+	assert.NoError(t, err)
+}
+
+func TestGMSAS3File(t *testing.T) {
+	// Setup required role
+	if os.Getenv("TEST_DISABLE_EXECUTION_ROLE") == "true" {
+		t.Skip("TEST_DISABLE_EXECUTION_ROLE was set to true")
+	}
+
+	// execution role arn is following the pattern arn:aws:iam::accountId:role/***
+	executionRole := os.Getenv("ECS_FTS_EXECUTION_ROLE")
+	if executionRole == "" {
+		t.Skip("ECS_FTS_EXECUTION_ROLE was not set to run this test")
+	}
+
+	// Setup s3 artifact
+	s3Bucket := os.Getenv("ECS_FTEST_S3_BUCKET")
+	s3BucketRegion := os.Getenv("ECS_FTEST_S3_BUCKET_REGION")
+	if s3Bucket == "" || s3BucketRegion == "" {
+		t.Skip("Skipping test as it requires both ECS_FTEST_S3_BUCKET and ECS_FTEST_S3_BUCKET_REGION to be set.")
+	}
+	t.Logf("Using s3 bucket %s in region %s", s3Bucket, s3BucketRegion)
+
+	sessWithRegion := session.Must(session.NewSession(aws.NewConfig().WithRegion(s3BucketRegion)))
+	svc := s3manager.NewUploaderWithClient(s3.New(sessWithRegion))
+
+	testCredSpecData := `{
+    "CmsPlugins":  [
+                       "ActiveDirectory"
+                   ],
+    "DomainJoinConfig":  {
+                             "Sid":  "S-1-5-21-975084816-3050680612-2826754290",
+                             "MachineAccountName":  "gmsa-acct-test",
+                             "Guid":  "92a07e28-bd9f-4bf3-b1f7-0894815a5257",
+                             "DnsTreeName":  "gmsa.test.com",
+                             "DnsName":  "gmsa.test.com",
+                             "NetBiosName":  "gmsa"
+                         },
+    "ActiveDirectoryConfig":  {
+                                  "GroupManagedServiceAccounts":  [
+                                                                      {
+                                                                          "Name":  "gmsa-acct-test",
+                                                                          "Scope":  "gmsa.test.com"
+                                                                      }
+                                                                  ]
+                              }
+}`
+
+	_, err := svc.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(s3Bucket),
+		Key:    aws.String("gmsa-test-file.json"),
+		Body:   strings.NewReader(testCredSpecData),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == errCodeAccessDenied {
+			t.Skip("Skipping the test since lacking necessary s3 access permission")
+		} else {
+			t.Fatalf("Unable to upload config file to s3 which is needed for the test: %v", err)
+		}
+	}
+
+	// Setup agent
+	RequireDockerAPIVersion(t, ">=1.39")
+
+	agentOptions := AgentOptions{
+		ExtraEnvironment: map[string]string{
+			"ZZZ_SKIP_DOMAIN_JOIN_CHECK_NOT_SUPPORTED_IN_PRODUCTION": "true",
+			"ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION":                  "1m",
+		},
+	}
+
+	agent := RunAgent(t, &agentOptions)
+	defer agent.Cleanup()
+
+	agent.RequireVersion(">=1.33.0")
```

Comment:
Same here. The version needs to be updated to 1.32.1

Reply:
Done, thanks!